File: src/favoringIsolatedMutability/java/create/UseHollywoodActor.java
/***
 * Excerpted from "Programming Concurrency on the JVM",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/vspcon for more book information.
***/
package com.agiledeveloper.pcj;

import akka.actor.ActorRef;
import akka.actor.Actors;

public class UseHollywoodActor {
  public static void main(final String[] args) throws InterruptedException { 
    final ActorRef johnnyDepp = Actors.actorOf(HollywoodActor.class).start();   
    johnnyDepp.sendOneWay("Jack Sparrow");
    Thread.sleep(100);
    johnnyDepp.sendOneWay("Edward Scissorhands");
    Thread.sleep(100);
    johnnyDepp.sendOneWay("Willy Wonka");
    Actors.registry().shutdownAll();
  }
}

File: bin/concurrency-code/favoringIsolatedMutability/scala/create/UseHollywoodActor.scala
package com.agiledeveloper.pcj

import akka.actor.{Actor, Actors, ActorRegistry}

//START:CODE
object UseHollywoodActor {
  def main(args : Array[String]) :Unit = { 
    val johnnyDepp = Actor.actorOf[HollywoodActor].start()    
    
    johnnyDepp ! "Jack Sparrow"
    Thread.sleep(100)
    johnnyDepp ! "Edward Scissorhands"
    Thread.sleep(100)
    johnnyDepp ! "Willy Wonka"

    Actors.registry.shutdownAll
  }
}
//END:CODE
